### PR TITLE
#3328 - Add bond custom query field

### PR DESCRIPTION
**Pull request by AKZhuk**
## How the feature works? / How did you fix the issue?

https://github.com/epam/ketcher/assets/16281184/01949c42-fb96-4c28-9761-08c2440d993b

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

### CODE REVIEW

File: packages/ketcher-core/src/domain/entities/bond.ts [hunk 1]
```diff
@@ -120,6 +124,13 @@ export class Bond {
 
     if (attributes.stereo) this.stereo = attributes.stereo;
     if (attributes.topology) this.topology = attributes.topology;
+    if (attributes.customQuery) {
+      this.customQuery = attributes.customQuery;
+      this.type = 8;
```
Comment (Nitvex) [inline packages/ketcher-core/src/domain/entities/bond.ts:129]: Could you, please, clarify, why `8`?

File: packages/ketcher-react/src/script/ui/data/convert/structconv.js [hunk 2]
```diff
@@ -277,18 +277,22 @@ function toRlabel(values) {
...
 }
 
 export function toBond(bond) {
+  const isCustomQuery = bond.customQuery !== '';
   return {
     topology: bond.topology,
     reactingCenterStatus: bond.center,
-    ...toBondType(bond.type),
+    customQuery: !isCustomQuery ? null : bond.customQuery,
+    ...toBondType(isCustomQuery ? 'any' : bond.type),
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/data/convert/structconv.js:295]: I believe, there should be something like:
```
...toBondType(isCustomQuery ? Bond.PATTERN.TYPE.ANY : bond.type),
```

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.module.less [hunk 3]
```diff
@@ -35,4 +35,37 @@
...
   }
+
+  [aria-haspopup="listbox"][aria-disabled="true"]{
+    cursor: not-allowed!important;
+    border: 1px solid @color-border-input!important;
+    background-color: @color-grey-3!important;
+  }
+}
+
+.customQueryWrapper{
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.module.less:46]:
```suggestion
.customQueryWrapper {
```

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx [hunk 4]
```diff
@@ -45,26 +93,79 @@ const Bond = (props: Props) => {
...
+    up: '/',
+    down: '\\',
+  };
+  if (type in bondType) {
+    queryAttrsText += bondType[type];
+  }
+  if (queryAttrsText) {
+    queryAttrsText += ';';
+  }
+  if (topology === 1) {
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx:165]: I propose to not use magic numbers

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx [hunk 5]
```diff
@@ -45,26 +93,79 @@ const Bond = (props: Props) => {
...
         />
         <Field
           name="center"
           component={Select}
           options={getSelectOptionsFromSchema(bondProps.center)}
+          disabled={isCustomQuery}
+          formName="bond-properties"
         />
+        <div className={classes.customQueryWrapper}>
+          <CustomQueryField
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx:124]: My proposal is to add data-testid for this in advance

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.module.less [hunk 6]
```diff
@@ -35,4 +35,37 @@
...
+  label{
```
Comment (OlgaMazurina) [inline packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.module.less:47]: Spaces are missed here and below as well

**Review by OlgaMazurina — APPROVED** [review]
Good job!

File: packages/ketcher-core/src/domain/entities/bond.ts (hunk 1 above)
Comment (AKZhuk) [reply packages/ketcher-core/src/domain/entities/bond.ts:129]: done

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx (hunk 4 above)
Comment (AKZhuk) [reply packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx:165]: done

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx (hunk 5 above)
Comment (AKZhuk) [reply packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.tsx:124]: done

File: packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.module.less (hunk 6 above)
Comment (AKZhuk) [reply packages/ketcher-react/src/script/ui/views/modal/components/toolbox/Bond/Bond.module.less:47]: done

File: packages/ketcher-react/src/script/ui/data/convert/structconv.js (hunk 2 above)
Comment (AKZhuk) [reply packages/ketcher-react/src/script/ui/data/convert/structconv.js:295]: decided with author not to fix it

**Pull request merged**